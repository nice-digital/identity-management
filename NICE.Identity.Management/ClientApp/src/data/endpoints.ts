const API_BASE_URL = process.env.REACT_APP_API_BASE_URL
	? process.env.REACT_APP_API_BASE_URL
	: "#{REACT_APP_API_BASE_URL}";

export const Endpoints = {
	usersList: `${API_BASE_URL}/users`, // get all users
	user: (userId: string | number): string => `${API_BASE_URL}/users/${userId}`,

	servicesList: `${API_BASE_URL}/services`,
	service: (serviceId: string | number): string =>
		`${API_BASE_URL}/services/${serviceId}`,
	
	userRolesByWebsite: (userId: string | number, websiteId: string | number): string =>
		`${API_BASE_URL}/users/${userId}/rolesbywebsite/${websiteId}`,

	verificationEmail: `${API_BASE_URL}/VerificationEmail/VerificationEmail`,
	websitesList: `${API_BASE_URL}/websites`,
};
