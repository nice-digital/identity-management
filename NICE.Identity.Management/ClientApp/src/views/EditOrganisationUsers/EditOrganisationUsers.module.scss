.orgUsersListSummary {
  font-size: 1.25rem;
  margin: 0;
}
