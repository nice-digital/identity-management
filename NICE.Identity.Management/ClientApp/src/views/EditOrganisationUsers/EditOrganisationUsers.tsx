import React, { Component } from "react";
import { RouteComponentProps, Link } from "react-router-dom";
import { Alert } from "@nice-digital/nds-alert";
import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
import { Grid, GridItem } from "@nice-digital/nds-grid";
import { PageHeader } from "@nice-digital/nds-page-header";
import { Table } from "@nice-digital/nds-table";
import { Endpoints } from "../../data/endpoints";
import { fetchData } from "../../helpers/fetchData";
import { isDataError } from "../../helpers/isDataError";
import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
import { UsersAndJobIdsByOrganisationType, UserType } from "../../models/types";
import { UserStatus } from "../../components/UserStatus/UserStatus";
import { FilterSuggestions } from "../../components/FilterSuggestions/FilterSuggestions";
import { SuggestionUser } from "../../components/SuggestionUser/SuggestionUser";
import styles from "./EditOrganisationUsers.module.scss";

type TParams = { id: string };

type EditOrganisationProps = Record<string, unknown> &
	RouteComponentProps<TParams>;

type EditOrganisationUsersState = {
	organisation: UsersAndJobIdsByOrganisationType;
	usersSearch: UserType[];
	confirmationMessage: string;
	isLoading: boolean;
	error?: Error;
};
export class EditOrganisationUsers extends Component<
	EditOrganisationProps,
	EditOrganisationUsersState
> {
	constructor(props: EditOrganisationProps) {
		super(props);

		this.state = {
			organisation: {} as UsersAndJobIdsByOrganisationType,
			usersSearch: [],
			confirmationMessage: "",
			isLoading: true,
		};

		document.title = "NICE Accounts - Edit organisation users";
	}

	async componentDidMount(): Promise<void> {
		this.setState({ isLoading: true });
		const organisationId = this.props.match.params.id;

		const organisation = await fetchData(
			Endpoints.usersAndJobIdsByOrganisation(organisationId),
		);

		if (isDataError(organisation)) {
			this.setState({ error: organisation });
		}

		this.setState({ organisation, isLoading: false });
	}

	handleRemoveUserClick = async (
		e: React.MouseEvent<HTMLAnchorElement>,
		jobId: number,
	): Promise<void> => {
		e.preventDefault();
		this.setState({ isLoading: true });

		const organisation = { ...this.state.organisation };

		const fetchOptions = {
			method: "DELETE",
			headers: { "Content-Type": "application/json" },
			body: JSON.stringify({
				id: jobId,
			}),
		};

		await fetchData(Endpoints.job(jobId), fetchOptions);

		organisation.users = organisation.users.filter(
			(orgUser) => orgUser.jobId !== jobId,
		);

		this.setState({
			organisation,
			confirmationMessage: "User has been successfully removed.",
			isLoading: false,
		});
	};

	filterUsersSearchList = async (
		searchQuery: string,
	): Promise<void | boolean> => {
		this.setState({ confirmationMessage: "" });

		if (!searchQuery.length) {
			this.setState({ usersSearch: [] });
			return false;
		}

		searchQuery = searchQuery.toLowerCase();

		let usersSearch = await fetchData(Endpoints.usersListSearch(searchQuery));

		if (isDataError(usersSearch)) {
			this.setState({ error: usersSearch });
		} else {
			const organisation = { ...this.state.organisation };
			const existingUsersIds = organisation.users.map((user) => user.userId);

			usersSearch = usersSearch.filter(
				(user: UserType) => !existingUsersIds.includes(user.userId),
			);
			usersSearch = usersSearch.filter(
				(user: UserType) =>
					`${user.firstName} ${user.lastName}`
						.toLowerCase()
						.indexOf(searchQuery) > -1 &&
					user.hasVerifiedEmailAddress &&
					!user.isLockedOut,
			);

			this.setState({ usersSearch });
		}
	};

	addUserToOrganisation = async (item: UserType): Promise<void | boolean> => {
		const itemIsEmpty = Object.keys(item).length === 0;

		if (itemIsEmpty) {
			this.setState({ usersSearch: [] });
			return false;
		}

		this.setState({ isLoading: true });

		const organisation = { ...this.state.organisation };

		const fetchOptions = {
			method: "POST",
			headers: { "Content-Type": "application/json" },
			body: JSON.stringify({
				userId: item.userId,
				organisationId: Number(this.props.match.params.id),
				isLead: false,
			}),
		};

		const job = await fetchData(Endpoints.jobs, fetchOptions);

		organisation.users = [
			...organisation.users,
			{ userId: item.userId, user: item, jobId: job.id },
		];

		this.setState({
			organisation,
			usersSearch: [],
			confirmationMessage: "User has been successfully added.",
			isLoading: false,
		});
	};

	render(): JSX.Element {
		const {
			organisation: { organisation, users },
			usersSearch,
			confirmationMessage,
			isLoading,
			error,
		} = this.state;
		const organisationId = this.props.match.params.id;

		return (
			<>
				<Breadcrumbs>
					<Breadcrumb to="/overview" elementType={Link}>
						Administration
					</Breadcrumb>
					<Breadcrumb to="/organisations" elementType={Link}>
						Organisations
					</Breadcrumb>
					<Breadcrumb
						to={`/organisations/${organisationId}`}
						elementType={Link}
					>
						{isLoading ? "Loading organisation details" : organisation?.name}
					</Breadcrumb>
					<Breadcrumb>Edit users</Breadcrumb>
				</Breadcrumbs>

				<PageHeader
					preheading={
						isLoading ? "Loading Organisation Name" : organisation?.name
					}
					heading="Users"
					lead="Add or remove users from this organisation"
				/>

				{!error ? (
					<>
						<Grid>
							<GridItem cols={12} md={3}>
								<FilterSuggestions
									label="Find and add an active user account; search excludes Pending and Locked accounts"
									data={usersSearch}
									qaSelExtract="add-organisation-user"
									onInputChange={this.filterUsersSearchList}
									onResultClick={this.addUserToOrganisation}
									elementType={SuggestionUser}
								/>
							</GridItem>
							<GridItem cols={12} md={9} aria-busy={isLoading}>
								{confirmationMessage && (
									<Alert
										type="info"
										role="status"
										aria-live="polite"
										data-qa-sel="added-edit-organisation-users"
									>
										<p>{confirmationMessage}</p>
									</Alert>
								)}
								<h2 className={styles.orgUsersListSummary}>Current users</h2>
								{isLoading ? (
									<p>Loading...</p>
								) : users.length ? (
									<Table
										style={{ display: "table" }}
										data-qa-sel="list-of-organisation-users"
									>
										<thead>
											<tr>
												<th>User name</th>
												<th>Email address</th>
												<th>Status</th>
												<th>Remove user</th>
											</tr>
										</thead>
										<tbody>
											{users.map((orgUser) => {
												const {
													user: { firstName, lastName, emailAddress },
												} = orgUser;

												return (
													<tr key={orgUser.userId}>
														<td>{`${firstName} ${lastName}`}</td>
														<td>{emailAddress}</td>
														<td>
															<UserStatus user={orgUser.user} />
														</td>
														<td>
															<a
																onClick={(e) =>
																	this.handleRemoveUserClick(
																		e,
																		orgUser.jobId as number,
																	)
																}
																href={`#remove-org-user`}
															>
																Remove user
															</a>
														</td>
													</tr>
												);
											})}
										</tbody>
									</Table>
								) : (
									<p>No users found for this organisation.</p>
								)}
							</GridItem>
						</Grid>
					</>
				) : (
					<ErrorMessage error={error}></ErrorMessage>
				)}
			</>
		);
	}
}
