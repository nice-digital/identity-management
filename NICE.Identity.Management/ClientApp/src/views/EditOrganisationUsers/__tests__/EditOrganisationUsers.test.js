import React from "react";
import { act } from "react-dom/test-utils";
import { render, waitForElementToBeRemoved, screen, waitFor, within } from "@testing-library/react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router";
import { EditOrganisationUsers } from "../EditOrganisationUsers";
import { Endpoints } from "src/data/endpoints";
import orgUsers from "./orgUsers.json";
import users from "./users.json";
import usersMore from "./usersMore.json";

const consoleErrorReset = console.error;

beforeEach(() => {
	console.error = consoleErrorReset;
});

const server = setupServer(
  rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
    return res(ctx.json(orgUsers));
  }),
  rest.get(Endpoints.usersList, (req, res, ctx) => {
    return res(ctx.json(users));
  }),
  rest.delete(Endpoints.job("1082"), (req, res, ctx) => {
    return res(ctx.json({}));
  }),
  rest.post(Endpoints.jobs, (req, res, ctx) => {
    return res(ctx.json({}));
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("should show loading message before data has been loaded", () => {
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
	render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	const loadingMessage = screen.getByText("Loading...", { selector: "p" });
	expect(loadingMessage).toBeInTheDocument();
});

test("should match the snapshot after data has been loaded", async () => {
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
	const {container} = render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
	expect(container).toMatchSnapshot();
});

test("should show error message when fetchData function returns 401 error", async () => {
	console.error = jest.fn();	
	const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
	server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(
				ctx.status(401),
				ctx.json({})
			);
		})
	);
	const {container} = render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	await waitFor(() => screen.getByRole("heading", { name: "Error"}));
	expect(container).toMatchSnapshot();
});

test("should show error message when fetchData function returns 500 error", async () => {
	console.error = jest.fn();	
	const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
	server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(
				ctx.status(500),
				ctx.json({})
			);
		})
	);
	const {container} = render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	await waitFor(() => screen.getByRole("heading", { name: "Error"}));
	expect(container).toMatchSnapshot();	
});

test("should show no users found message when an organisation has no users assigned", async () => {
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
  const orgUsersEmpty = {
    organisationId: 1,
    organisation: { "id": 1, "name": "NICE", "dateAdded": null },
    usersAndJobIds: [],
  };
	server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(
				ctx.json(orgUsersEmpty)
			);
		})
	);
	render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	const noUsersFoundMessage = await screen.findByText("No users found for this organisation.");
  expect(noUsersFoundMessage).toBeInTheDocument();
});

test("should add user to table when suggestion is clicked and show confirmation message", async () => {
  jest.useFakeTimers();
  console.error = jest.fn();
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  }; 
  const orgUsersEmpty = {
    organisationId: 1,
    organisation: { "id": 1, "name": "NICE", "dateAdded": null },
    usersAndJobIds: [],
  };
  server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(ctx.json(orgUsersEmpty));
		})
	);
  render(<MemoryRouter><EditOrganisationUsers match={match} /></MemoryRouter>);
  await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
  const filterSearchInput = screen.getByRole("searchbox");
  filterSearchInput.focus();
  await act(async () => {
		userEvent.paste("kri")
		jest.advanceTimersByTime(1000);	
  });
  let suggestion = await screen.findByText("Kristin Patrick", { selector: "a" });
  expect(suggestion).toBeInTheDocument();
  userEvent.click(suggestion);
  const userAdded = await screen.findByText("Kristin Patrick", { selector: "td" });
  expect(userAdded).toBeInTheDocument();
  expect(screen.getByText("User has been successfully added.", { selector: "p" })).toBeInTheDocument();
  jest.useFakeTimers();
});

test("should remove user from org and show confirmation message when selected from table", async () => {
  console.error = jest.fn();	
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };  
	render(<EditOrganisationUsers match={match} />, {wrapper: MemoryRouter});
	await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
  const user = userEvent.setup();
	let removeUserLinks = screen.getAllByRole("link", { name: "Remove user" });
  const lastRemoveUserLink = removeUserLinks[removeUserLinks.length - 1];
  expect(removeUserLinks.length).toEqual(4);
  user.click(lastRemoveUserLink);
  const userRemovedMessage = await screen.findByText("User has been successfully removed.");
  removeUserLinks = screen.getAllByRole("link", { name: "Remove user" });
  expect(userRemovedMessage).toBeInTheDocument();
  expect(removeUserLinks.length).toEqual(3);
});

test("should show 3 suggestions when search term has been entered", async () => {
  jest.useFakeTimers();
  console.error = jest.fn();
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };
  const orgUsersEmpty = {
    organisationId: 1,
    organisation: { "id": 1, "name": "NICE", "dateAdded": null },
    usersAndJobIds: [],
  };
	server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(ctx.json(orgUsersEmpty));
		}),
    rest.get(Endpoints.usersList, (req, res, ctx) => {
			return res.once(ctx.json(usersMore));
		}),
	);
  render(<MemoryRouter><EditOrganisationUsers match={match} /></MemoryRouter>);
  await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
  const filterSearchInput = screen.getByRole("searchbox");
  filterSearchInput.focus();
  await act(async () => {
		userEvent.paste("kri")
		jest.advanceTimersByTime(1000);	
  });
  await waitFor(() => screen.getByText("Kristin Patrick"));
  const suggestionsList= await screen.findByRole("list", { name: "suggestions-add-organisation-user"});
	const { queryAllByRole } = within(suggestionsList);
	const suggestionsListItems = queryAllByRole("listitem");
  expect(suggestionsListItems.length).toEqual(3);
  jest.useFakeTimers();
});

test("should hide suggestions when clicked off", async () => {
  jest.useFakeTimers();
  console.error = jest.fn();
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  };  
  render(<MemoryRouter><EditOrganisationUsers match={match} /></MemoryRouter>);
  await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
  const filterSearchInput = screen.getByRole("searchbox");
  filterSearchInput.focus();
  await act(async () => {
		userEvent.paste("kri")
		jest.advanceTimersByTime(1000);	
  });
  let suggestion = await screen.findAllByText("Kristin Patrick");
  expect(suggestion.length).toEqual(1);
  userEvent.tab();
  suggestion = screen.queryAllByText("Kristin Patrick");
  expect(suggestion.length).toEqual(0);
  jest.useFakeTimers();
});

test("should hide existing confirmation/alert when suggestions search has been initiated again", async () => {
  jest.useFakeTimers();
  console.error = jest.fn();
  const match = {
    params: { id: 1 },
    isExact: true,
    path: "",
    url: "",
  }; 
  const orgUsersEmpty = {
    organisationId: 1,
    organisation: { "id": 1, "name": "NICE", "dateAdded": null },
    usersAndJobIds: [],
  };
  server.use(
		rest.get(Endpoints.usersAndJobIdsByOrganisation("1"), (req, res, ctx) => {
			return res.once(ctx.json(orgUsersEmpty));
		})
	);
  render(<MemoryRouter><EditOrganisationUsers match={match} /></MemoryRouter>);
  await waitForElementToBeRemoved(() => screen.getByText("Loading...", { selector: "p" }));
  const filterSearchInput = screen.getByRole("searchbox");
  filterSearchInput.focus();
  await act(async () => {
		userEvent.paste("kri")
		jest.advanceTimersByTime(1000);	
  });
  let suggestion = await screen.findByText("Kristin Patrick", { selector: "a" });
  expect(suggestion).toBeInTheDocument();
  userEvent.click(suggestion);
  const userAddedMessage = await screen.findByText("User has been successfully added.", { selector: "p" });
  expect(userAddedMessage).toBeInTheDocument();
  filterSearchInput.focus();
  await act(async () => {
		userEvent.paste("a")
		jest.advanceTimersByTime(1000);	
  });
  expect(userAddedMessage).not.toBeInTheDocument();
  jest.useFakeTimers();
});