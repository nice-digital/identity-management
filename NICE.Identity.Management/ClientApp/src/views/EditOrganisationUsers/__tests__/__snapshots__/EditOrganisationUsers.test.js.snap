// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot after data has been loaded 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/overview"
        >
          Administration
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations"
        >
          Organisations
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations/1"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Edit users
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/overview",
        "name": "Administration"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/organisations",
        "name": "Organisations"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/organisations/1",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Edit users"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
  >
    <h1
      class="page-header__heading"
    >
      <span
        class="page-header__pre-heading"
      >
        NICE
      </span>
      Users
    </h1>
    <p
      class="page-header__lead"
    >
      Add or remove users from this organisation
    </p>
  </div>
  <div
    class="grid"
  >
    <div
      data-g="12 md:3"
    >
      <div
        class="panel"
      >
        <div
          class="input noMargin"
        >
          <label
            class="input__label"
            for="filter-suggestions"
          >
            Find and add an active user account; search excludes Pending and Locked accounts
          </label>
          <input
            autocomplete="off"
            class="input__input"
            data-qa-sel="filter-suggestions-input"
            id="filter-suggestions"
            name="filter-suggestions"
            type="search"
            value=""
          />
        </div>
        <div
          class="suggestions"
        >
          <ul
            aria-label="suggestions-add-organisation-user"
          />
        </div>
      </div>
    </div>
    <div
      aria-busy="false"
      data-g="12 md:9"
    >
      <h2
        class="orgUsersListSummary"
      >
        Current users
      </h2>
      <table
        class="table"
        data-qa-sel="list-of-organisation-users"
        style="display: table;"
      >
        <thead>
          <tr>
            <th>
              User name
            </th>
            <th>
              Email address
            </th>
            <th>
              Status
            </th>
            <th>
              Remove user
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Jonny Russell
            </td>
            <td>
              jonathan.russell@nice.org.uk
            </td>
            <td>
              <span
                class="tag tag--consultation"
                data-qa-sel="user-status"
              >
                Active
              </span>
            </td>
            <td>
              <a
                href="#remove-org-user"
              >
                Remove user
              </a>
            </td>
          </tr>
          <tr>
            <td>
              Catherine Quigley
            </td>
            <td>
              catherine.quigley@nice.org.uk
            </td>
            <td>
              <span
                class="tag tag--consultation"
                data-qa-sel="user-status"
              >
                Active
              </span>
            </td>
            <td>
              <a
                href="#remove-org-user"
              >
                Remove user
              </a>
            </td>
          </tr>
          <tr>
            <td>
              Andy Child
            </td>
            <td>
              andrew.child@nice.org.uk
            </td>
            <td>
              <span
                class="tag tag--consultation"
                data-qa-sel="user-status"
              >
                Active
              </span>
            </td>
            <td>
              <a
                href="#remove-org-user"
              >
                Remove user
              </a>
            </td>
          </tr>
          <tr>
            <td>
              Colin Mazurkiewicz
            </td>
            <td>
              colin.mazurkiewicz@nice.org.uk
            </td>
            <td>
              <span
                class="tag tag--consultation"
                data-qa-sel="user-status"
              >
                Active
              </span>
            </td>
            <td>
              <a
                href="#remove-org-user"
              >
                Remove user
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`should show error message when fetchData function returns 401 error 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/overview"
        >
          Administration
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations"
        >
          Organisations
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations/1"
        />
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Edit users
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/overview",
        "name": "Administration"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/organisations",
        "name": "Organisations"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/organisations/1"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Edit users"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
  >
    <h1
      class="page-header__heading"
    >
      Error
    </h1>
  </div>
  <div
    class="alert alert--error"
    role="alert"
  >
    <h2>
      Whoops
    </h2>
    <p>
      Sorry, there's been an error!
    </p>
  </div>
</div>
`;

exports[`should show error message when fetchData function returns 500 error 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/overview"
        >
          Administration
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations"
        >
          Organisations
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/organisations/1"
        />
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Edit users
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/overview",
        "name": "Administration"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/organisations",
        "name": "Organisations"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/organisations/1"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Edit users"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
  >
    <h1
      class="page-header__heading"
    >
      Error
    </h1>
  </div>
  <div
    class="alert alert--error"
    role="alert"
  >
    <h2>
      Whoops
    </h2>
    <p>
      Sorry, there's been an error!
    </p>
  </div>
</div>
`;
